rust/operator-binary/src/controller.rs

```diff
@@ -562,22 +562,30 @@ pub async fn reconcile_nifi(
             )
             .await?;
 
-            if resolved_product_image.product_version.starts_with("1.") {
-                let rg_metrics_service = build_rolegroup_metrics_service(
-                    nifi,
-                    &rolegroup,
-                    role_group_service_recommended_labels,
-                    role_group_service_selector.into(),
-                )
-                .context(ServiceConfigurationSnafu)?;
+            // Determine the service ports to expose
+            // 1.X.X: extra metrics ports
+            // 2.X.X: via NiFi HTTP(s) port
+            let service_ports = if resolved_product_image.product_version.starts_with("1.") {
+                metrics_service_ports()
+            } else {
+                headless_service_ports()
```

Comment:
Currently this does the right thing. However, conceptually it should *only* be the https port (even if more ports are added in the future), I would prefer if the function name expresses this.